Transformer (#17)
neuralhydrology/neuralhydrology: docs/source/usage/config.rst

   current input timescale. In both cases, ``transfer_mtslstm_states``
   can be used to configure hidden and cell state transfer.

Transformer settings

> ~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
~~~~~~~~~~~~~~~~~~~~
```
just to avoid annoying warnings ("Title underline too short") in the docs building process

These are used if ``model == transformer``.

-  ``transformer_nlayers``: Number of multi-head self-attention layers in the 
   transformer encoder.

> -  ``transformer_positional_encoding_type``: Choices are ``[add, concat]``.
>    Used to change the way that the positional encoding is used in transformer
>    embedding layer.

Reviewer 2: ```suggestion
-  ``transformer_positional_encoding_type``: Choices are ``[add, concat]``.
   Used to change the way that the positional encoding is used in transformer
   embedding layer. `add` means that the positional encoding is added to the values
   of the inputs for that layer, while `concat` means that the embedding is concatenated
   as additional input features.
```

   transformer encoder.

> -  ``transformer_positional_encoding_type``: Choices are ``[add, concat]``.

Reviewer: ```suggestion
-  ``transformer_positional_encoding_type``: Choices are ``[sum, concatenate]``.
```
inconsistent with implementation.

   Used to change the way that the positional encoding is used in transformer

>    embedding layer. `add` means that the positional encoding is added to the values
>    of the inputs for that layer, while `concat` means that the embedding is concatenated

Reviewer: ```suggestion
   embedding layer. `sum` means that the positional encoding is added to the values
   of the inputs for that layer, while `concatenate` means that the embedding is concatenated
```
